File: programmers/1845.py
def solution(nums):
    answer = 0
    set_nums = set(nums)
    answer = len(nums)//2 if len(set_nums) > len(nums)//2 else len(set_nums)

    return answer


solution([3, 1, 2, 3])
solution([3, 3, 3, 2, 2, 4])
solution([3, 3, 3, 2, 2, 2])

File: programmers/1845.js
function solution(nums) {
	var answer = 0;

	let cnt = new Set(nums);
	// console.log(cnt);
	if (cnt.size >= nums.length / 2) {
		answer = nums.length / 2;
	} else {
		answer = cnt.size;
	}

	console.log(answer);
	return answer;
}
